libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationEventDispatcherInterface.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.services.android.navigation.v5.internal.navigation.metrics.NavigationMetricListener
+import com.mapbox.services.android.navigation.v5.location.RawLocationListener
+import com.mapbox.services.android.navigation.v5.milestone.Milestone
+import com.mapbox.services.android.navigation.v5.milestone.MilestoneEventListener
+import com.mapbox.services.android.navigation.v5.navigation.EnhancedLocationListener
+import com.mapbox.services.android.navigation.v5.navigation.NavigationEventListener
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteListener
+import com.mapbox.services.android.navigation.v5.route.FasterRouteListener
+import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+
+interface NavigationEventDispatcherInterface {
```

Comment:
@olegzil it's a really huge interface, would you like to split it to smallest? it stays more readable, of course up 2 u.

Reply:
If you look at the comment "// Interface boundary" you'll see the proposal as to where this interface can be partitioned.